### PR TITLE
feat(roles): add roles to list-users endpoint

### PR DESCRIPTION
**Pull request by epigos**
# Description

Please include a summary of the change and which issue is fixed.

If you are a Gr4vy employee, please also reference the related Zoho tickets.
If you do not have a related Zoho ticket and this is more than a bug fix, then
create one now.

[TA-3206](https://gr4vy.atlassian.net/browse/TA-3206)

- Return `roles` assigned to the user for the following user endpoints

  - `GET /users`
  - `POST /users`
  - `GET /users/:id`
  - `GET /users/me`
  - `POST /users/reset-password`

- Include `roles` query parameter to `GET /users` endpoint to enable filtering results by role IDs

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own changes
- [ ] I have run `yarn test` to make sure my changes pass all linters
- [ ] I have pulled the latest changes from the upstream `main` branch

## Contribution guidelines

For contribution guidelines, styleguide, and other helpful information please
see the `CONTRIBUTING.md` file in the root of this project.

### CODE REVIEW

File: reference/resources/user.yaml [hunk 1]
```diff
@@ -52,3 +52,9 @@ properties:
     format: date-time
     example: "2013-07-16T19:23:00.000+00:00"
     nullable: true
+
+  roles:
+    type: array
+    description: A list of roles.
```
Comment (cbetta) [inline reference/resources/user.yaml:58]:
```suggestion
    description: The roles this user has been assigned.
```

File: reference/query-parameters/roles.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+---
+schema:
+  type: string
+  example: 8724fd24-5489-4a5d-90fd-0604df7d3b83
+in: query
+name: roles
+description: |-
+  Filters the results to only match `user` by the `roles`
+  assigned to them. It support multiple role IDs.
```
Comment (cbetta) [inline reference/query-parameters/roles.yaml:9]:
```suggestion
schema:
  type: array
  items:
    type: string
  example: 
    - 8724fd24-5489-4a5d-90fd-0604df7d3b83
in: query
name: roles
description: |-
  Filters the results to only match the users that have any `roles`
  that matches with any of the provided role IDs.
```

Comment (cbetta) [reply reference/query-parameters/roles.yaml:9]: This follows the format we use here as well https://github.com/gr4vy/gr4vy-openapi/blob/main/reference/query-parameters/list_transactions__status.yaml

**Review by cbetta — CHANGES_REQUESTED** [review]

File: reference/query-parameters/roles.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+---
+schema:
+  type: array
+  items:
+    type: string
+  example:
```
Comment (cbetta) [inline reference/query-parameters/roles.yaml:6]:
```suggestion
  example:
```

**Review by thomas-gr4vy — APPROVED** [review]

**Review by cbetta — APPROVED** [review]

**Pull request merged**